src/app/pages/products-list/pagination-control/pagination-control.component.ts

```diff
@@ -0,0 +1,35 @@
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+
+@Component({
+    selector: 'app-pagination-control',
+    templateUrl: './pagination-control.component.html',
+    styleUrls: ['./pagination-control.component.css'],
+})
+export class PaginationControlComponent {
+    @Input() pageIndexes: number[] | null = null;
+    @Input() activeIndex: number | null = null;
+
+    @Output() selectIndex = new EventEmitter<number>();
+    @Output() back = new EventEmitter<void>();
+    @Output() next = new EventEmitter<void>();
+
+    onSelectIndex(index: number): void {
+        this.selectIndex.emit(index);
+    }
+
+    onBack(): void {
+        this.back.emit();
+    }
+
+    onNext(): void {
+        this.next.emit();
+    }
+
+    isActiveIndex(index: number): boolean {
+        if (this.activeIndex === null) {
+            return false;
+        }
+
+        return this.activeIndex === index;
```

Comment:
Похожие конструкции, в которых по какому-то значению определяется какие данные возвращать, отлично укладывается в тернарный оператор:
```suggestion
        return this.activeIndex === null ? false : this.activeIndex === index;
```

А можно пойти еще дальше и воспользоваться вычислением значения через логический оператор `&&`:
```suggestion
        return this.activeIndex !== null && this.activeIndex === index;
```
OR
```suggestion
        const isActiveIndexDefined = this.activeIndex !== null;
        
        return isActiveIndexDefined && this.activeIndex === index;
```

Reply:
Ок